contracts/serverless-v2/CommonChainJobs.sol

```diff
@@ -0,0 +1,357 @@
+    }
+
+    function _authorizeUpgrade(
+        address /*account*/
+    ) internal view override onlyAdmin {}
+
+    //-------------------------------- Overrides end --------------------------------//
+
+    //-------------------------------- Initializer start --------------------------------//
+
+    function __CommonChainJobs_init(
+        address _admin,
+        IERC20 _token,
+        CommonChainGateways _gateways,
+        CommonChainExecutors _executors,
+        uint256 _relayBufferTime,
+        uint256 _executionBufferTime,
+        uint256 _noOfNodesToSelect
+    ) public initializer {
+        require(_admin != address(0), "ZERO_ADDRESS_ADMIN");
+
+        __Context_init();
+        __ERC165_init();
+        __AccessControlEnumerable_init();
+        __UUPSUpgradeable_init();
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+
+        token = _token;
+        gateways = _gateways;
+        executors = _executors;
+        relayBufferTime = _relayBufferTime;
+        executionBufferTime = _executionBufferTime;
+        noOfNodesToSelect = _noOfNodesToSelect;
+    }
+
+    //-------------------------------- Initializer end --------------------------------//
+
+    IERC20 public token;
+    CommonChainGateways public gateways;
+    CommonChainExecutors public executors;
+    uint256 public relayBufferTime;
+    uint256 public executionBufferTime;
+    uint256 public noOfNodesToSelect;
+
+    function setGatewaysContract(CommonChainGateways _gateways) external onlyAdmin {
+        gateways = _gateways;
+    }
+
+    function setExecutorsContract(CommonChainExecutors _executors) external onlyAdmin {
+        executors = _executors;
+    }
+
+    function setTokenContract(IERC20 _token) external onlyAdmin {
+        token = _token;
+    }
+
+    //-------------------------------- Job start --------------------------------//
+
+    struct Job {
+        uint256 jobId;
+        uint256 reqChainId;
+        bytes32 codehash;
+        bytes codeInputs;
+        uint256 deadline;
+        uint256 execStartTime;
+        address jobOwner;
+        address gatewayOperator;
+        uint8 outputCount;
+        uint8 sequenceId;
+        bool isResourceUnavailable;
+    }
+
+    // jobKey => Job
+    mapping(uint256 => Job) public jobs;
+
+    // jobKey => executors
+    mapping(uint256 => address[]) public selectedExecutors;
+    // jobKey => selectedExecutorAddress => hasExecuted
+    mapping(uint256 => mapping(address => bool)) public hasExecutedJob;
+
+    event JobRelayed(
+        uint256 indexed jobId,
+        uint256 indexed reqChainId,
+        bytes32 codehash,
+        bytes codeInputs,
+        uint256 deadline,
+        address jobOwner,
+        address gatewayOperator,
+        address[] selectedExecutors
+    );
+
+    event JobResponded(
+        uint256 indexed jobId,
+        uint256 indexed reqChainId,
+        bytes output,
+        uint256 totalTime,
+        uint256 errorCode,
+        uint8 outputCount
```

Comment:
Can you add `gatewayOperator` in this event as well? I need to in the gateway enclave to verify which gateway should process this event initially.